src/AssociatedJobInfoClient.cs

```diff
@@ -0,0 +1,124 @@
+using Microsoft.DotNet.HelixPoolProvider.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.HelixPoolProvider
+{
+    public class AssociatedJobInfo
+    {
+        public string BuildSourceBranch { get; }
+        public string SystemPullRequestTargetBranch { get; }
+
+        public static readonly AssociatedJobInfo Empty
+            = new AssociatedJobInfo("", "");
+
+        public AssociatedJobInfo(
+            string buildSourceBranch,
+            string systemPullRequestTargetBranch)
+        {
+            BuildSourceBranch = buildSourceBranch;
+            SystemPullRequestTargetBranch = systemPullRequestTargetBranch;
+        }
+    }
+
+    public class AssociatedJobInfoClient
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly ILogger<AssociatedJobInfoClient> _logger;
+
+        public AssociatedJobInfoClient(
+            IHttpClientFactory httpClientFactory,
+            ILogger<AssociatedJobInfoClient> logger)
+        {
+            _httpClientFactory = httpClientFactory;
+            _logger = logger;
+        }
+
+        public async Task<AssociatedJobInfo> TryGetAssociatedJobInfo(
```

Comment:
Without a timeout of maximum 10 seconds for this entire operation, this can't be checked in. As others have said, the entire pool provider request can't go over 30 seconds, and that includes time to create a whole helix job which has in the past taken more than the whole 30 seconds. Event 10 seconds for this operation feels long, and knowing the azdo apis it will probably take longer than that.

Reply:
I have set the timeout to 10 seconds.